### PR TITLE
sesh: init at 0.12.0

### PR DESCRIPTION
**Pull request by gwg313**
## Description of changes

Sesh is a Smart session manager for the terminal that can be to assist with tmux workflows
https://github.com/joshmedeski/sesh

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/se/sesh/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "joshmedeski";
+    repo = "sesh";
+    rev = "v${version}";
+    sha256 = "sha256-m/EcWh4wfna9PB/NN+MCRUsz5Er0OZ70AAumlKdrm/s=";
+  };
+
+  vendorHash = "sha256-zt1/gE4bVj+3yr9n0kT2FMYMEmiooy3k1lQ77rN6sTk=";
+
+  ldflags = ["-s" "-w"];
```
Comment (AndersonTorres) [inline pkgs/by-name/se/sesh/package.nix:19]:
```suggestion
  ldflags = [ "-s" "-w" ];
```

File: pkgs/by-name/se/sesh/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  meta = {
+    description = "Smart session manager for the terminal";
+    homepage = "https://github.com/joshmedeski/sesh";
+    changelog = "https://github.com/joshmedeski/sesh/releases/tag/${src.rev}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [gwg313];
```
Comment (AndersonTorres) [inline pkgs/by-name/se/sesh/package.nix:26]:
```suggestion
    maintainers = with lib.maintainers; [ gwg313 ];
```

File: pkgs/by-name/se/sesh/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  vendorHash = "sha256-zt1/gE4bVj+3yr9n0kT2FMYMEmiooy3k1lQ77rN6sTk=";
+
+  ldflags = ["-s" "-w"];
+
+  meta = {
+    description = "Smart session manager for the terminal";
+    homepage = "https://github.com/joshmedeski/sesh";
+    changelog = "https://github.com/joshmedeski/sesh/releases/tag/${src.rev}";
+    license = lib.licenses.mit;
```
Comment (AndersonTorres) [inline pkgs/by-name/se/sesh/package.nix:25]: Add a `mainProgram`.

File: pkgs/by-name/se/sesh/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}:
+buildGoModule rec {
+  pname = "sesh";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "joshmedeski";
+    repo = "sesh";
+    rev = "v${version}";
+    sha256 = "sha256-m/EcWh4wfna9PB/NN+MCRUsz5Er0OZ70AAumlKdrm/s=";
```
Comment (AndersonTorres) [inline pkgs/by-name/se/sesh/package.nix:14]:
```suggestion
    hash = "sha256-m/EcWh4wfna9PB/NN+MCRUsz5Er0OZ70AAumlKdrm/s=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/se/sesh/package.nix (hunk 3 above)
Comment (gwg313) [reply pkgs/by-name/se/sesh/package.nix:25]: done

**Review by AndersonTorres — APPROVED** [review]

**Comment by itslychee** [comment]
Result of `nixpkgs-review pr 286594` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sesh</li>
  </ul>
</details>

**Review by itslychee — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by JulienMalka — APPROVED** [review]
Looks good to me. Before merging could you reword your commit message to remove all the intermediate steps that were added in there ?

**Pull request merged**